packages/terra-functional-testing/src/reporters/file-output-reporter/index.js

```diff
@@ -0,0 +1,207 @@
+const fs = require('fs-extra');
+const path = require('path');
+const stripAnsi = require('strip-ansi');
+const SpecReporter = require('@wdio/spec-reporter').default;
+const endOfLine = require('os').EOL;
+const { Logger } = require('@cerner/terra-cli');
+
+const LOG_CONTEXT = '[terra-functional-testing:file-output-reporter]';
+
+class FileOutputReporter extends SpecReporter {
+  constructor() {
+    super();
+    this.runners = [];
+    this.resultJsonObject = {
+      startDate: '',
+      type: 'wdio',
+      locale: '',
+      formFactor: '',
+      theme: '',
+      output: {},
+      endDate: '',
+    };
+    this.fileName = '';
+    this.moduleName = process.cwd().split(path.sep).pop();
+
+    this.setResultsDir = this.setResultsDir.bind(this);
+    this.hasResultsDir = this.hasResultsDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.printReport = this.printReport.bind(this);
+    this.getMessage = this.getMessage.bind(this);
+
+    this.setResultsDir();
+    this.hasResultsDir();
+  }
+
+  /**
+   * Sets results directory for the test run. Uses the wdio reporterOptions.outputDir if set, otherwise
+   * it outputs to tests/wdio/reports.
+   * @return null;
+   */
+  setResultsDir() {
+    this.resultsDir = path.join(process.cwd(), 'tests', 'wdio', 'reports');
+  }
+
+  /**
+   * Check and create reports dir if doesn't exist
+   * @return null
+   */
+  hasResultsDir() {
+    if (!fs.existsSync(this.resultsDir)) {
+      fs.mkdirSync(this.resultsDir, { recursive: true }, (err) => {
+        if (err) {
+          Logger.error(err.message, { context: LOG_CONTEXT });
+        }
+      });
+    }
+  }
+
+  /**
+   * Formatting the filename based on locale, theme, and formFactor
+   * @return null
+   */
+  fileNameCheck({ formFactor, locale, theme }, browserName, cid) {
+    const fileNameConf = ['fileOutput'];
+    if (locale) {
+      fileNameConf.push(locale);
+      this.resultJsonObject.locale = locale;
+    }
+
+    if (theme) {
+      fileNameConf.push(theme);
+      this.resultJsonObject.theme = theme;
+    }
+
+    if (formFactor) {
+      fileNameConf.push(formFactor);
+      this.resultJsonObject.formFactor = formFactor;
+    }
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  setTestModule(specsValue) {
+    const index = specsValue.lastIndexOf(`packages${path.sep}`);
+    if (index > -1) {
+      const testFilePath = specsValue.substring(index).split(path.sep);
+      const moduleName = testFilePath && testFilePath[1]
+        ? testFilePath[1]
+        : process.cwd().split(path.sep).pop();
```

Comment:
There was issue #597 in the spec reporter where the package name was incorrect when running on Jenkins CI. A change had to be made to parse the package name correctly. Does this potentially have the same issue that needs to be taken care of? This was the [fix](https://github.com/cerner/terra-toolkit/pull/613/files#diff-c4330edcf41e36606925cf520272bc65282b2454fee1d8d50c8c131ecc042ee6R123-R127).

cc @BenBoersma

Reply:
If `setTestModule` is doing the same thing as [getPackageName](https://github.com/cerner/terra-toolkit/blob/main/packages/terra-functional-testing/src/reporters/spec-reporter/wdio-spec-reporter.js#L118-L129), would it be easier to copy the logic over and set it to `this.moduleName`?

Reply:
I just realized setTestModule and getPackageName are doing two different things. setTestModule is grabbing the package name what it is running. where as getPackageName is getting the repo name